code/service/src/main/scala/com/namely/chiefofstate/ServiceMigrationRunner.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Namely Inc.
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+package com.namely.chiefofstate
+
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import com.namely.chiefofstate.migration.{DbUtil, JdbcConfig, Migrator}
+import com.namely.chiefofstate.migration.versions.v1.V1
+import com.namely.chiefofstate.migration.versions.v2.V2
+import com.namely.chiefofstate.migration.versions.v3.V3
+import com.namely.chiefofstate.serialization.{MessageWithActorRef, ScalaMessage}
+import com.namely.protobuf.chiefofstate.v1.internal.{DoMigration, MigrationDone}
+import com.typesafe.config.Config
+import org.slf4j.{Logger, LoggerFactory}
+import slick.basic.DatabaseConfig
+import slick.jdbc.JdbcProfile
+
+import scala.util.{Failure, Success, Try}
+
+/**
+ * kick starts the various migrations needed to run.
+ * When the migration process is successful it replies back to the [[ServiceBootstrapper]] to continue the boot process.
+ * However when the migration process fails it stops and the [[StartNodeBehaviour]] get notified and halt the whole
+ * boot process. This is the logic behind running the actual migration
+ * <p>
+ *   <ol>
+ *     <li> check the existence of the cos_migrations table
+ *     <li> if the table exists go to step 4
+ *     <li> if the table does not exist run the whole migration and reply back to the [[ServiceBootstrapper]]
+ *     <li> check the current version against the available versions
+ *     <li> if current version is the last version then no need to run the migration just reply back to [[ServiceBootstrapper]]
+ *     <li> if not then run the whole migration and reply back to the [[ServiceBootstrapper]]
+ *   </ol>
+ * </p>
+ */
+object ServiceMigrationRunner {
+  final val log: Logger = LoggerFactory.getLogger(getClass)
+
+  def apply(config: Config): Behavior[ScalaMessage] = Behaviors
+    .setup[ScalaMessage] { context =>
+      Behaviors.receiveMessage[ScalaMessage] {
+        case MessageWithActorRef(message, replyTo) =>
```

Comment:
```suggestion
        case MessageWithActorRef(DoMigration(), replyTo) =>
```
 You can remove a level of nesting